dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_configurations_panel.tsx

```diff
@@ -219,22 +260,32 @@ export const DataConfigPanelItem = ({
     (visualizations.vis.name === VIS_CHART_TYPES.Line ||
       visualizations.vis.name === VIS_CHART_TYPES.Scatter);
 
+  const getTimeStampFilteredFields = (options: IField[]) =>
+    filter(options, (i) => i.type !== TIMESTAMP);
+
   const getOptionsAvailable = (sectionName: string) => {
     const selectedFields = {};
     const unselectedFields = fieldOptionList.filter((field) => !selectedFields[field.label]);
-    return sectionName === AGGREGATIONS
-      ? unselectedFields
-      : visualizations.vis.name === VIS_CHART_TYPES.Line ||
-        visualizations.vis.name === VIS_CHART_TYPES.Scatter
-      ? unselectedFields.filter((i) => i.type === 'timestamp')
-      : sectionName === BREAKDOWNS
-      ? configList[GROUPBY]
-      : unselectedFields;
+    if (sectionName === AGGREGATIONS) return unselectedFields;
+    if (sectionName === BREAKDOWNS) return configList[GROUPBY];
+    if (
+      visualizations.vis.name === VIS_CHART_TYPES.Line ||
+      visualizations.vis.name === VIS_CHART_TYPES.Scatter
+    )
+      return filter(unselectedFields, (i) => i.type === TIMESTAMP);
+    if (!isTimeStampSelected && configList.span?.time_field.length > 0)
+      return getTimeStampFilteredFields(unselectedFields);
+    if (
+      (selectedConfigItem.name === GROUPBY && selectedConfigItem.index === 0) ||
+      isTimeStampSelected
+    )
+      return unselectedFields;
```

Comment:
Make sure to check all conditions are working properly. You might have already done so, but make sure we do not skip any conditions here during testing.

Reply:
Yes checked it